### PR TITLE
Prevent Duplicate Event Creation #29

### LINKED ISSUE #29: Prevent Duplicate Event Creation
Duplicate events should be prevented.

If name, country, city, street, start date (not time) and end date (not time) should not be the same.

### PR DESCRIPTION
**Pull request by t-gul**
- Duplicate rule and matching rule is created for "Public Event"
- "Event Id" field is created under "Public Event" 
- "Before Insert Update Public Event" flow is created for fill the "Event Id"

### CODE REVIEW

**Review by bmtemakina — APPROVED** [review]

File: force-app/main/default/profiles/Admin.profile-meta.xml [hunk 1]
```diff
@@ -146,6 +146,11 @@
         <field>Event_Registration__c.Registration_Id__c</field>
```
Comment (sercanEmakina) [inline force-app/main/default/profiles/Admin.profile-meta.xml:146]: @sgoEma do you have any idea why there isn't a flowaccess tag in Admin profile for "Before_Insert_Update_Public_Event" flow that is newly created and included into this PR?

Comment (t-gul) [reply force-app/main/default/profiles/Admin.profile-meta.xml:146]: Yes, "Before_Insert_Update_Public_Event" flow that is newly created and included into this PR
I'm  checking Admin Profile now

Comment (t-gul) [reply force-app/main/default/profiles/Admin.profile-meta.xml:146]: Admin Profile is updated.

File: manifest/package.xml [hunk 2]
```diff
@@ -92,6 +97,14 @@
 		<members>Session__c.All</members>
 		<name>ListView</name>
 	</types>
+	<types>
```
Comment (sgoEma) [inline manifest/package.xml:101]: @t-gul one of the matching rule types can be removed I think :/ One of them should work properly. Can you test and leave the one that works?

File: force-app/main/default/flows/Before_Insert_Update_Public_Event.flow-meta.xml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Flow xmlns="http://soap.sforce.com/2006/04/metadata">
+    <apiVersion>54.0</apiVersion>
+    <description>Create unique Id and set Event Id with it</description>
+    <formulas>
+        <name>CreatedEventId</name>
+        <dataType>String</dataType>
+        <expression>LOWER({!$Record.Name})+LOWER({!$Record.Country__c})+LOWER({!$Record.City__c})+TEXT(DATEVALUE({!$Record.Start_Date__c}))+TEXT(DATEVALUE({!$Record.End_Date__c}))</expression>
```
Comment (sgoEma) [inline force-app/main/default/flows/Before_Insert_Update_Public_Event.flow-meta.xml]: @t-gul As I remember we have decided to split only the date fields into fields. Name, Country, City should be matched seperately in duplicate/matching rule. I was expecting date fields only within this pull request

**Review by sgoEma — CHANGES_REQUESTED** [review]
Instead of Event Id, we need start/end dates on the object. Dates, Name, Country, and City should be checked on Duplicate/Matching Rule seperately. If you collect all values in one field, changing it to the unique field is easier than duplicate rule creation. Here we want to apply another solution using duplicate rules.

File: manifest/package.xml (hunk 2 above)
Comment (t-gul) [reply manifest/package.xml:101]: both of them work, I'm erase one of them

**Review by sgoEma — APPROVED** [review]

**Review by sercanEmakina — APPROVED** [review]

**Pull request merged**